src/pages/workspaces/workspace/Data.js

```diff
@@ -526,6 +530,171 @@ const EntitiesContent = ajaxCaller(class EntitiesContent extends Component {
   }
 })
 
+const DeleteObjectModal = ajaxCaller(class DeleteObjectModal extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { deleting: false }
+  }
+
+  async delete() {
+    const { name, workspace: { workspace: { namespace, bucketName } }, ajax: { Buckets }, onSuccess } = this.props
+    try {
+      this.setState({ deleting: true })
+      await Buckets.delete(namespace, bucketName, name)
+      onSuccess()
+    } catch (error) {
+      reportError('Error deleting object', error)
+    } finally {
+      this.setState({ deleting: false })
+    }
+  }
+
+  render() {
+    const { onDismiss } = this.props
+    const { deleting } = this.state
+    return h(Modal, {
+      onDismiss,
+      okButton: () => this.delete(),
+      title: 'Are you sure you want to delete this file?'
```

Comment:
Does this action delete the file from the Google Bucket? If so, we should tell users that.